proposals/slottable-request.md

```diff
@@ -0,0 +1,86 @@
+# Slottable Request Protocol
+aka "Render Props for Web Components"
+
+**Author**: Kevin Schaaf (@kevinpschaaf, Google)
+
+**Status**: Draft
+
+**Last Updated**: 2023-08-28
+
+
+# Background
+
+There are often use cases that require a component to render UI based on data, but where the specific rendering of the data should be controllable by the user. Use cases of this include:
+
+*   A data table or tree component, which manages the rendering of cells or tree nodes, but where the given look and feel of the items is customizable by the user.
+*   A virtual list component, which optimizes rendering a list of data by only rendering a subset of user-templated items based on which array instances are visible on the screen.
```

Comment:
This virtual list does not use this pattern: https://github.com/WICG/virtual-scroller

Worth reviewing why they came to the decision not to.

Reply:
The virtual-scroller effort is defunct, but even so it did seem to follow a _very_ similar pattern where the container fires an event to tell the host to render new content: https://github.com/WICG/virtual-scroller#rangechange-event

Can you say what you think the important differences are?

Reply:
It's unclear to me how the event-listener is intended to provide the content. Inline examples would help clarify this for me.